File: algorithms/507.cpp
#include <iostream>
#include <cmath>
using namespace std;

class Solution {
public:
    bool checkPerfectNumber(int num) {
        if (num < 2)
            return false;
        int sum = 1;
        for (int i = 2; i < sqrt(num); i++)
            if (num%i == 0)
                sum += i + num/i;
        return sum == num;
    }
};

int main() {
    return 0;
}

File: algorithms/python3/507.py
class Solution:
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 1:
            return False
        s, t = 1, int(num ** 0.5)
        for i in range(2, t+1):
            if num % i == 0:
                s += i + num//i
        if t * t == num:
            s -= t
        return s == num
